lib/_007/Q.pm

```diff
@@ -209,18 +206,43 @@ class Q::Term::Object does Q::Term {
     has $.propertylist;
 
     method eval($runtime) {
+        my $type = $runtime.get-var($.type.name.value, $.type.frame);
+        # XXX: Need to do a subtyping check here instead
+        if $type === TYPE<Exception> {
```

Comment:
`.isa` should be perfect here

Reply:
Yes, except `.isa` needs an instance, and here we only have a type. I think the shortest way to do the check currently is

    if TYPE<Exception> (elem) $type.type-chain {

Reply:
Heh. I just removed the comment for now. The reason is that the code in question is in high flux and will settle down to something completely different (and simpler) once the refactor winds down. The comment was basically adorning a hack that didn't deserve it.